### PR TITLE
ENG-2043 Fix "Build all" bug where failures still count as success

### PR DESCRIPTION
**Pull request by tsiq-kylel**
Build-all on master (even if one of the intermediates fail) will return exit code 0.

This could eventually cause a problem on travis if users use :all instead of build, push, then genDockerrun separately.

### CODE REVIEW

File: src/main/java/com/twosigmaiq/AllGoalsMojo.java [hunk 1]
```diff
@@ -75,9 +77,18 @@ public void execute() throws MojoExecutionException {
     setupInvokerLogger(invoker);
 
     try {
-      invoker.execute(invocationRequest);
+      InvocationResult invocationResult = invoker.execute(invocationRequest);
+      if (invocationResult.getExitCode() != 0) {
+        String msg = "Invocation Exception";
```
Comment (tsiq-hanxie) [inline src/main/java/com/twosigmaiq/AllGoalsMojo.java]: Could use ternary here instead to make it more succinct:

```
String msg = invocationResult.getExecutionException() != null ? invocationResult.getExecutionException().getMessage(): "Invocation Exception";
```

File: src/main/java/com/twosigmaiq/AllGoalsMojo.java [hunk 2]
```diff
@@ -75,9 +77,18 @@ public void execute() throws MojoExecutionException {
     setupInvokerLogger(invoker);
 
     try {
-      invoker.execute(invocationRequest);
+      InvocationResult invocationResult = invoker.execute(invocationRequest);
```
Comment (tsiq-echeung) [inline src/main/java/com/twosigmaiq/AllGoalsMojo.java]: It's good practice to make vars finals when you don't expect them to be assigned new values.

File: src/main/java/com/twosigmaiq/AllGoalsMojo.java [hunk 3]
```diff
@@ -75,9 +77,18 @@ public void execute() throws MojoExecutionException {
...
+      if (invocationResult.getExitCode() != 0) {
+        String msg = "Invocation Exception";
+        if (invocationResult.getExecutionException() != null) {
+          msg = invocationResult.getExecutionException().getMessage();
+        }
+        throw new CommandLineException(msg);
+      }
     } catch (MavenInvocationException e) {
```
Comment (tsiq-echeung) [inline src/main/java/com/twosigmaiq/AllGoalsMojo.java]: If you have catch multiple exceptions, and handle them the same way, put them in the same catch statement.

ie. "catch (MavenInvocationException | CommandLineException e)"

File: src/main/java/com/twosigmaiq/AllGoalsMojo.java (hunk 1 above)
Comment (tsiq-kylel) [reply src/main/java/com/twosigmaiq/AllGoalsMojo.java]: tried to do this, though I keep getting Checkstyle validations.

Comment (tsiq-hanxie) [reply src/main/java/com/twosigmaiq/AllGoalsMojo.java]: kk not a problem then

**Pull request merged**